lib/regexp/regexp.go

```diff
@@ -0,0 +1,290 @@
+//
+//     regexp.find_all(src, max) - Returns a new, mutable list of all successive matches of the regular expression regexp. An empty list indicates
+//                                 no match. If max > 0, at most max strings are returned. If max == 0,
+//                                 an empty list is returned. If max < 0, all strings are returned.
+//                                 The parameter max is optional: by default no limit is applied.
+//
+//     regexp.find_submatches(src) - Returns a new, mutable list of strings holding the text of the leftmost match of the regular expression regexp
+//                                   in the given string and the matches, if any, of its subexpressions. An empty list indicates
+//                                   no match.
+//
+//     regexp.matches(src) - Reports whether the given string contains any match of the regular expression regexp.
+//
+//     regexp.replace_all(src, repl) - Returns a copy of the given string, replacing matches of the regular expression regexp
+//                                     with the replacement string repl. Inside repl, backslash-escaped digits (\1 to \9) can be
+//                                     used to insert text matching corresponding parenthesized group from the pattern.
+//                                     \0 in repl refers to the entire matching text.
+//
+//     regexp.replace_all(src, replFunc) - Returns a copy of the given string in which all matches of the regular expression regexp
+//                                         have been replaced by the return value of the replacement function applied to the
+//                                         matched substring.
+//
+//     regexp.split(src, max) - Returns a new, mutable list of strings between all the matches of the regular expression regexp.
+//                              If max > 0, at most max strings are returned knowing that the last string is
+//                              the unsplit remainder. If max == 0, an empty list is returned. If max < 0,
+//                              all strings are returned. The parameter max is optional: by default no limit
+//                              is applied.
+//
+var Module = &starlarkstruct.Module{
+	Name: "regexp",
+	Members: starlark.StringDict{
+		"compile": starlark.NewBuiltin("compile", compile),
+	},
+}
+
+// The regular expression used to indentify all the backreferences in a replacement pattern.
+var backreferenceRe = regexp.MustCompile(`((\\\\)*)\\(\d)`)
+
+// The regular expression used to identify the forbidden patterns.
+var forbiddenPatternRe = regexp.MustCompile(`([^\\]|^)(\\\\)*\\C`)
+
+func compile(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		pattern string
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &pattern); err != nil {
+		return nil, err
+	}
+
+	if forbiddenPatternRe.MatchString(pattern) {
```

Comment:
Now that I think clearly, the Go implementation of RE2 already rejects \C for us (because it too is a rune-oriented not byte-oriented machine). So we can delete this check. Sorry for wasting your time. (The regexp was neat though!)

Reply:
@adonovan I knew that it was not supported but as you wanted a clear error message, I added it but ok let's remove it